src/main/java/de/retest/recheck/review/ignore/AttributeRegexFilter.java

```diff
@@ -48,9 +51,26 @@ protected AttributeRegexFilterLoader( final Pattern regex ) {
 		}
 
 		@Override
-		protected Optional<AttributeRegexFilter> load( final MatchResult regex ) {
-			final String attributeRegex = regex.group( 1 );
-			return Optional.of( new AttributeRegexFilter( attributeRegex ) );
+		public String save( final Filter ignore ) {
+			if ( ignore instanceof AllMatchFilter ) {
+				return new AllMatchFilterLoader().save( (AllMatchFilter) ignore );
+			}
+			return super.save( ignore );
+		}
+
+		@Override
+		protected Optional<Filter> load( final MatchResult regex ) {
+			String match = regex.group( 1 );
+			if ( !match.contains( "," ) ) {
+				return Optional.of( new AttributeRegexFilter( match ) );
+			}
+			final String remainder = match.substring( match.indexOf( ',' ) + 1 ).trim();
+			match = match.substring( 0, match.indexOf( ',' ) );
+			// TODO Either no optional as return or no exception below
+			final Filter chained = AttributeFilterLoader.chainableFilter.load( remainder ). //
```

Comment:
We share `chainableFilter`, but the remainder of the `load` method is essentially the same (besides simple vs. regex filter).

Does it make sense to share more? Or KISS over DRY? In case of the latter, I think I would also duplicate `chainableFilter`.

My preferred way of doing it, something like:

```java
class ChainableFilterLoaderUtil {

	private static final Loader<Filter> chainableFilter = new InheritanceLoader<>( Arrays.asList( //
			Pair.of( PixelDiffFilter.class, new PixelDiffFilterLoader() ) //
	) );

	public static Optional<Filter> load( final MatchResult regex, final Function<String, Filter> simpleFilterLoader ) {
		final String match = regex.group( 1 );
		if ( !match.contains( "," ) ) {
			return Optional.of( simpleFilterLoader.apply( match ) );
		}
		final String prefix = match.substring( 0, match.indexOf( ',' ) );
		final String suffix = match.substring( match.indexOf( ',' ) + 1 ).trim();
		return chainableFilter.load( suffix ) //
				.map( chained -> new AllMatchFilter( simpleFilterLoader.apply( prefix ), chained ) );
	}

}
```

Then this could be:

```java
@Override
protected Optional<Filter> load( final MatchResult regex ) {
	return ChainableFilterLoaderUtil.load( regex, match -> new AttributeRegexFilter( match ) );
}
```

Reply:
Great idea, will try!